diesel/src/query_builder/select_statement/dsl_impls.rs

```diff
@@ -483,11 +519,35 @@ where
             self.limit_offset,
             self.group_by,
             self.locking,
+            self.having,
+        )
+    }
+}
+
+impl<F, S, D, W, O, LOf, G, LC, H, Predicate> HavingDsl<Predicate>
+    for SelectStatement<F, S, D, W, O, LOf, G, LC, H>
```

Comment:
I'm not sure if `HAVING` clauses are valid for all select statements (like for such with locking clauses). Also `HAVING` clauses are basically `WHERE` clauses for cases where no `GROUP BY` is set, so we probably want to allow them just for queries which already have a `GROUP BY` clause set.

Reply:
I removed the generic from the implementation of `LockingDsl`, as it was the only implementation which had the `GROUP BY` generic default to `NoGroupByClause`. 

I'm unsure - should i remove the generic for `HAVING` from some of the other impls, or would this be fine? 
I'm assuming that if the `HAVING` clause doesn't work with some of the other clauses, it should also be the case that neither does `GROUP BY` and/or `WHERE` 🤔 

Also added a trait bound to `HavingDsl` to verify that there is a `GROUP BY` set.

Reply:
> Also added a trait bound to HavingDsl to ensure that there is a GROUP BY set.

I would rather like to do that via setting the generic parameter `G` to `GroupByClause<Expr>` here, also set `LC` to `NoLockingClause` to prevent the usage with an locking clause.


```suggestion
    for SelectStatement<F, S, D, W, O, LOf, GroupByClause<G>, NoLockingClause, H>
```